lib/IDE/CompletionLookup.cpp

```diff
@@ -2017,9 +2021,10 @@ void CompletionLookup::foundDecl(ValueDecl *D, DeclVisibilityKind Reason,
 
     if (auto *GP = dyn_cast<GenericTypeParamDecl>(D)) {
       addGenericTypeParamRef(GP, Reason, dynamicLookupInfo);
-      for (auto *protocol : GP->getConformingProtocols())
-        addConstructorCallsForType(protocol->getDeclaredInterfaceType(),
-                                   GP->getName(), Reason, dynamicLookupInfo);
+      auto type =
+          CurrDeclContext->mapTypeIntoContext(GP->getDeclaredInterfaceType());
+      addConstructorCallsForType(type, GP->getName(), Reason,
+                                 dynamicLookupInfo);
```

Comment:
@bnbarham Could you re-review? I added this to fix an issue for generic parameters:
```swift
func foo<MyGeneric: MyProtocol>(_: MyGeneric) {
  <HERE>
}
```
In this case, since this is a `GenericTypeParamDecl`, the type passed for `addConstructorCallsForType()` was a protocol type which is an existential type. But in this case we do want initializers for `MyGeneric`.